lib/iris/_concatenate.py

```diff
@@ -146,6 +146,9 @@ def __new__(cls, coord, dims):
                                                       kwargs)
         return metadata
 
+    def __hash__(self):
+        return hash(tuple(self))
```

Comment:
They are, but when you override `__eq__`, you must supply `__hash__` as well. Unfortunately, you can't do something like `__hash__ = namedtuple.__hash__` because `namedtuple` is actually a function and the real class is hidden.